From b14a382ad0d6d49aceed8e363a18efe4826a0209 Mon Sep 17 00:00:00 2001
From: Sergei Trofimovich <slyfox@gentoo.org>
Date: Sun, 19 Jan 2020 23:12:06 +0000
Subject: [cpio: PATCH] src/global.c: drop duplicate definition of 'program_name'

Noticed as a build failure against fresh gcc-master:
```
  CCLD     cpio
ld: ../gnu/libgnu.a(progname.o):(.bss+0x0):
  multiple definition of `program_name'; global.o:cpio/src/global.c:188: first defined here
collect2: error: ld returned 1 exit status
```

gcc-10 will change the default from -fcommon to fno-common:
https://gcc.gnu.org/PR85678.

The error also happens if CFLAGS=-fno-common passed explicitly.

* src/global.c: drop 'program_name' definition.

Bug: https://bugs.gentoo.org/705900
Signed-off-by: Sergei Trofimovich <slyfox@gentoo.org>
---
 src/global.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/src/global.c b/src/global.c
index fb3abe9..acf92bc 100644
--- a/src/global.c
+++ b/src/global.c
@@ -184,9 +184,6 @@ unsigned int warn_option = 0;
 /* Extract to standard output? */
 bool to_stdout_option = false;
 
-/* The name this program was run with.  */
-char *program_name;
-
 /* A pointer to either lstat or stat, depending on whether
    dereferencing of symlinks is done for input files.  */
 int (*xstat) ();
-- 
2.25.0

